app/navigation/root-navigator.tsx

```diff
@@ -275,10 +271,18 @@ export const RootStack: NavigatorType = () => {
       })
   }, [])
 
-  useEffect(
-    () => messaging().onTokenRefresh(() => token && addDeviceToken(client)),
-    [client, token],
-  )
+  useEffect(() => {
```

Comment:
Previously we have called `addDeviceToken` at the same time that we log in or save a new token, this caused an issue however because it was using the old version of the apolloclient expecting the new token to be present however the new token isn't present until a rerender. This moves the `addDeviceToken` into a lifecycle that is dependent on the client.